app/views/layouts/application.html.erb

```diff
@@ -23,6 +23,31 @@
   </head>
 
   <body>
+    <header>
+      <div class="logo">Book Exchange</div>
+      <div class="menu-toggle" onclick="toggleMenu()">
+          <div></div>
+          <div></div>
+          <div></div>
+      </div>
+      <nav>
+          <a href="#">Home</a>
+          <a href="#">About</a>
+          <a href="#">How It Works</a>
+          <a href="#">Sign Up</a>
+          <a href="#">Languages</a>
+      </nav>
+    </header>
     <%= yield %>
+    <footer>
+      <p>&copy; 2024 Book Exchange Service. All rights reserved. | <a href="#">Privacy Policy</a> | <a href="#">Terms of Service</a></p>
+    </footer>
+
+  <script>
+    function toggleMenu() {
```

Comment:
Can we try toggle without css ?

Reply:
I tried to do this without CSS but I had trouble making it work properly. Maybe better to do it using Bootstrap since it already has the same functionality?

Reply:
Yea, good idea. I see `data-bs-toggle="collapse"` will do that.
Not sure how to include bootstrap's js code into the project

Reply:
Sorry, I was thinking to use less js , maybe to try with stimulus
```
<div data-controller="menu">
    <div class="menu-toggle" data-menu-action="toggleMenu">
          <div></div>
          <div></div>
          <div></div>
      </div>
      <nav data-menu-target="menu">
      </nav>
</div>
```
```
// app/assets/javascript/controllers/menu_controller.js
import { Controller } from "stimulus"

export default class extends Controller {
  static targets = [ "menu" ]

  toggleMenu {
    this.menuTarget.classList.toggle('active');
  }
}
```

Reply:
I managed to do this with Bootstrap